src/views/FieldMappingModal.vue

```diff
@@ -0,0 +1,206 @@
+
+          <ion-item>
+            <ion-label>{{ $t("Ordered quantity") }}</ion-label>
+            <ion-select interface="popover" v-if="content.length" :placeholder = "$t('Select')" v-model="fieldMappings.quantity">
+              <ion-select-option :key="index" v-for="(prop, index) in Object.keys(content[0])">{{ prop }}</ion-select-option>
+            </ion-select>
+          </ion-item>
+
+          <ion-item>
+            <ion-label>{{ $t("Facility ID") }}</ion-label>
+            <ion-select interface="popover" v-if="content.length" :placeholder = "$t('Select')" v-model="fieldMappings.facility">
+              <ion-select-option :key="index" v-for="(prop, index) in Object.keys(content[0])">{{ prop }}</ion-select-option>
+            </ion-select>
+          </ion-item>
+      </ion-list>
+    </div>
+    <ion-fab vertical="bottom" horizontal="end" slot="fixed">
+      <ion-fab-button @click="setFieldMapping()">
+        <ion-icon :icon="saveOutline" />
+      </ion-fab-button>
+    </ion-fab>
+  </ion-content>
+</template>
+
+<script lang="ts">
+import { 
+  IonButtons,
+  IonButton,
+  IonContent,
+  IonHeader,
+  IonIcon,
+  IonTitle,
+  IonToolbar,
+  IonLabel,
+  IonItem,
+  IonList,
+  IonSelect,
+  IonSelectOption,
+  modalController
+} from "@ionic/vue";
+import { defineComponent } from "vue";
+import { close, save, saveOutline } from "ionicons/icons";
+import { useStore, mapGetters } from "vuex";
+import { UserService } from "@/services/UserService";
+import { hasError } from '@/utils'
+import { DateTime } from 'luxon';
+import { showToast } from '@/utils';
+import { translate } from "@/i18n";
+
+export default defineComponent({
+  name: "FieldMappingModal",
+  data() {
+    return {
+      mappingName: "",
+      queryString: '',
+      filteredTimeZones: [],
+      timeZones: [],
+      timeZoneId: '',
+      fieldMappings: {}
+    }
+  },
+  props: ["content", "fieldMapping"],
+  computed: {
+    ...mapGetters({
+      fieldMappings: 'user/getFieldMappings'
+    })
+  },
+  mounted() {
+    this.fieldMappings = this.fieldMapping;
+  },
+  methods: {
+    generateUniqueMappingPrefId() {
+      const id = Math.floor(Math.random() * 1000);
+      return !this.fieldMappings[id] ? id : this.generateUniqueMappingPrefId();
+    },
+    closeModal() {
+      modalController.dismiss({ dismissed: true });
+    },
+    escapeRegExp(text: string) {
+      //TODO Handle it in a better way
+      // Currently when the user types special character as it part of Regex expressions it breaks the code
+      // so removed the characters for now
+      return text.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&');
+    },
+    findTimeZone() { 
+      const regularExp = new RegExp(`${this.escapeRegExp(this.queryString)}`, 'i');
+      this.filteredTimeZones = this.timeZones.filter((timeZone: any) => {
+        return regularExp.test(timeZone.id) || regularExp.test(timeZone.label);
+      });
+    },
+    async getAvailableTimeZones() {
```

Comment:
I think there is no need to for the timeZone specific code in here.

Reply:
Sure sir.